Detectors/TPC/simulation/src/Digitizer.cxx

```diff
@@ -71,93 +71,86 @@ DigitContainer *Digitizer::Process(TClonesArray *points)
   signalArray.resize(nShapedPoints);
 
   static size_t hitCounter=0;
-  for(auto pointObject : *points) {
-#ifdef TPC_GROUPED_HITS
-    auto *inputgroup = static_cast<LinkableHitGroup*>(pointObject);
-    const int MCTrackID = inputgroup->GetTrackID();
-    for(size_t hitindex = 0; hitindex<inputgroup->getSize(); ++hitindex){
-      ElementalHit eh = inputgroup->getHit(hitindex);
+  for(auto& inputgroup : *hits) {
+    //    auto *inputgroup = static_cast<LinkableHitGroup*>(pointObject);
+    const int MCTrackID = inputgroup.GetTrackID();
+    for(size_t hitindex = 0; hitindex < inputgroup.getSize(); ++hitindex){
+      ElementalHit eh = inputgroup.getHit(hitindex);
```

Comment:
Shouldn't this be a `const auto& eh = inputgroup.getHit(hitindex)`.
In addition, I guess the `inputpoint` can now be removed altogether to clean up the code below, or do I overlook something?

Reply:
will take a look